05_testing_and_ci/intro_slides.md

```diff
@@ -0,0 +1,142 @@
+---
+type: slide
+slideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+    font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+</style>
+
+# Introduction to Testing
+
+---
+
+## Learning goals of the chapter
+
+- Understanding why and how to write tests
+- Understanding the different types of tests and their relevance in testing scientific software
+- Becoming familiar with Python testing frameworks `pytest` and `unittest`
+
+---
+
+## Lets start with a survey
+
+- survey link
+- slido ID
+
+---
+
+## What is testing?
+
+- Smelling old milk before using it!
+- A way to determine if a software is not producing reliable results and if so, what is the reason.
+- Manual testing vs. Automated testing.
+
+---
+
+## Why should you test your software?
+
+- Improve software reliability and reproducibility.
+- Make sure that unrelated changes (bugfixes, new features) do not affect other parts of software.
+- Generally all software is better off being tested regularly. Possible exceptions are very small codes with single users.
+- Ensure that a distributed version of a software actually works.
+
+---
+
+## Nomenclature in software testing
+
+- **Fixture**: preparatory set for testing
+- **Actual result**: what the code produces when given the fixture
+- **Expected result**: what the actual result is compared to
+- **Test coverage**: how much of the code does a test touch
+
+---
+
+## Some ways to test software
+
+- Assertions
+- Unit testing
+- Integration testing
+- Regression testing
+
+---
+
+## Assertions
+
+- Principle of *defensive programming*.
+- Python does nothing when an assertion is true; returns error when false.
```

Comment:
This is true for any language, no? But I would say "throw error" instead of "return error".